src/physical/aggregate.rs

```diff
@@ -343,3 +343,94 @@ impl_max_accumulator!(u8, UInt8);
 impl_max_accumulator!(u16, UInt16);
 impl_max_accumulator!(u32, UInt32);
 impl_max_accumulator!(u64, UInt64);
+
+pub struct First {}
+
+impl Aggregate for First {
+    fn create_accumulator(&self, input_type: &DataType) -> Box<dyn Accumulator> {
+        match input_type {
+            DataType::Int8 => Box::new(FirstAccumulator::<i8> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
+            DataType::Int16 => Box::new(FirstAccumulator::<i16> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
+            DataType::Int32 => Box::new(FirstAccumulator::<i32> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
+            DataType::Int64 => Box::new(FirstAccumulator::<i64> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
+            DataType::UInt8 => Box::new(FirstAccumulator::<u8> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
+            DataType::UInt16 => Box::new(FirstAccumulator::<u16> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
+            DataType::UInt32 => Box::new(FirstAccumulator::<u32> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
+            DataType::UInt64 => Box::new(FirstAccumulator::<u64> { values_order: VecDeque::new(), values_counts: BTreeMap::new() }),
```

Comment:
Please also make it work for DataType::Utf8, DataType::Timestamp and DataType:Struct.

They're used in group_by.rs so you can look there for an example. (Utf8 requires a seperate macro unfortunately, as you have to do .as_str() instead of .clone() on the value you append to the builder)

Reply:
Oh, and Boolean, as well as Null.

Reply:
You can ignore Struct for now, because implementing that as a map key could be hard.

Reply:
You can omit structs for now TBH, because creating a builder for them could be hard.

Reply:
I don't understand thata macro builder stuff with UTF8 and also the generic types for null and utf8, so I left them commented maybe you can do something with it :<